python/pyarrow/array.pxi

```diff
@@ -2743,6 +2766,36 @@ cdef class StructArray(Array):
         result.validate()
         return result
 
+    def sort(self, order="ascending", fieldname=None, **options):
+        """
+        Sort the StructArray
+
+        Parameters
+        ----------
+        order : str, default "ascending"
+            Which order to sort values in.
+            Accepted values are "ascending", "descending".
+        fieldname : str or None, default None
```

Comment:
>> We currently don't use "fieldname" anywhere. Maybe just `field` as the keyword argument?
>
> `field` implied a `Field` instance to me, that's why I went for `fieldname`.

Good point. Another suggestion, maybe something like `by=` ? 

(I suppose that in the future we might want to allow sorting by multiple fields?)